Added timeout parameter for API objects

ccstolley opened the pull request:
By default there is no timeout on requests to the InstagramAPI. This can
cause unexpected behavior if a system is not anticipating a request to
block indefinitely.

This patch stops short of imposing a default timeout value, but does
expose the ability to specify a timeout value upon creating an api object.

The README has also been updated.

Commit:
Added timeout parameter for API objects

By default there is no timeout on requests to the InstagramAPI. This can
cause unexpected behavior if a system is not anticipating a request to
block indefinitely.

This patch stops short of imposing a default timeout value, but does
expose the ability to specify a timeout value upon creating an api object.

The README has also been updated.

dellsystem commented:
:+1: for this change. I've personally been setting the default socket timeout to 5 seconds but being able to set a timeout on a specific InstagramAPI instance would be solid.

MrYoda commented:
:+1: please, merge this for next release

Commit:
Fix signed requests.

This is a rebased version of Kevin's original fix from here:
https://github.com/facebookarchive/python-instagram/pull/195/files

Commit:
Bump httplib2 version to get important fixes.

Commit:
Fix signed requests for tag searches. Bump version.

This includes a hack to get around urlencoding unicode tags.

Commit:
Merge pull request #1 from ccstolley/master

ok

Commit:
Fix unicode encoding bug.
